toolkit/scripts/toolchain/download_toolchain_rpm.sh

```diff
@@ -129,16 +168,39 @@ function download() {
 
             echo "SUCCESS" >> "$log_file"
             touch "$dst_file"
-            return
+            return 0
         else
             echo "Failed to download toolchain RPM: $rpm_name" >> "$attempt_log_file"
             echo "FAILURE" >> "$log_file"
         fi
     done
 
-    exit 1
+    return 1
+}
+
+function validate_signatures() {
+    echo "Validating toolchain RPM: $rpm_name" | tee -a "$log_file"
+
+    for key in $allowable_gpg_keys; do
+        echo "Adding key ($key) to empty workdir at ($work_dir)" >> "$log_file"
+        rpmkeys --root "$work_dir" --import "$key" >> "$log_file"
+    done
+
+    if ! rpmkeys --root "$work_dir" --checksig --verbose "$dst_file" -D "%_pkgverify_level signature" >> "$log_file"; then
+        echo "Failed to validate toolchain package $rpm_name signature, aborting." | tee -a "$log_file"
+        return 1
+    fi
+    return 0
 }
 
 mkdir -p "$(dirname "$log_file")"
 
-download
+if ! download; then
+    exit 1
+fi
```

Comment:
Unnecessary with `set -e` at the top:
```suggestion
download
```